### ISSUE TITLE
Python packages install in "local" subdir with external python, it affects spack bootstrap black

### ISSUE THREAD

**Issue by Jordan474** [LLNL/spack#41448]
### Steps to reproduce

TL;DR: With external python (>= 3.8), python packages files are installed in `<prefix>/local/bin`, `<prefix>/local/lib/site-packages...` etc.

I had a long-time issue with spack style on my ubuntu 22.04:

```sh
$ spack --backtrace style -t black ./lib/spack/spack/main.py
==> Warning: Cannot find any Python path in /home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd/view
==> Running style checks on spack
  selected: black
==> Modified files
  lib/spack/spack/main.py
==> Running black checks
Traceback (most recent call last):
  File "/home/me/git/github.com/spack/spack/bin/spack", line 52, in <module>
    sys.exit(main())
  File "/home/me/git/github.com/spack/spack/lib/spack/spack_installable/main.py", line 37, in main
    sys.exit(spack.main.main(argv))
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/main.py", line 1066, in main
    return _main(argv)
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/main.py", line 1019, in _main
    return finish_parse_and_run(parser, cmd_name, args, env_format_error)
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/main.py", line 1049, in finish_parse_and_run
    return _invoke_command(command, parser, args, unknown)
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/main.py", line 648, in _invoke_command
    return_val = command(parser, args)
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/cmd/style.py", line 422, in style
    return_code |= run_function(which(tool_name), file_list, args)
  File "/home/me/git/github.com/spack/spack/lib/spack/spack/cmd/style.py", line 346, in run_black
    output = black_cmd(*packed_args, fail_on_error=False, output=str, error=str)
TypeError: 'NoneType' object is not callable
```

I decided to dig into it, and after a couple layers deep, **I find `black` was not installed as `bin/black`, but as `local/bin/black`**:

```console
$ ls -l /home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd/view/bin/black
ls: cannot access '/home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd/view/bin/black': No such file or directory
$ ls -l /home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd/view/local/bin/black
lrwxrwxrwx 1 me me 136 déc.   6 12:15 /home/me/.spack/bootstrap/environments/python3.10-x86_64-7d8fd/view/local/bin/black -> /home/me/.spack/bootstrap/store/linux-ubuntu22.04-x86_64/gcc-12.3.0/py-black-23.1.0-zv2nmcjc7se7vl4ii7bo3tduebd6nywx/local/bin/black
```

After more testing, the problem seems to affect all python packages installed with an external python.

I thought of a bad configuration on my machine, but **I was able to reproduce it with a simple `py-pip` install and external python on a `ubuntu:22.04` docker image:**

```sh
set -e

apt-get -yqq update
apt-get -yqq install build-essential ca-certificates coreutils curl environment-modules gfortran git gpg lsb-release python3 python3-distutils python3-venv unzip zip

git clone https://github.com/spack/spack.git
cd spack
source share/spack/setup-env.sh

spack external find --scope=site python
cat etc/spack/packages.yaml

spack install py-pip
spack find --format '{prefix}' py-pip
ls -la $(spack find --format '{prefix}' py-pip)
```

at the end I get:

```console
[...]
==> py-pip: Successfully installed py-pip-23.1.2-7ocfxvsxrbf36dup2lzz5fj35j7cym6n
  Stage: 0.63s.  Install: 1.23s.  Post-install: 0.07s.  Total: 1.95s
[+] /home/me/git/github.com/spack/foobar/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/py-pip-23.1.2-7ocfxvsxrbf36dup2lzz5fj35j7cym6n
/home/me/git/github.com/spack/foobar/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/py-pip-23.1.2-7ocfxvsxrbf36dup2lzz5fj35j7cym6n
total 16
drwxr-sr-x 4 root root 4096 Dec  6 11:26 .
drwxr-xr-x 3 root root 4096 Dec  6 11:26 ..
drwxr-sr-x 3 root root 4096 Dec  6 11:27 .spack
drwxr-sr-x 4 root root 4096 Dec  6 11:27 local
```

---

Digging deeper I find the Python package `platlib` returns `local` in the path:

```sh
$ spack external find --scope=site python
python@2.7.18  python@3.10.12
$ spack python -c 'from spack.spec import Spec; s = Spec("python"); s.concretize(); print(s.prefix); print(s.package.platlib);'
/usr
local/lib/python3.10/dist-packages
$ python3 -c 'import sysconfig; print(sysconfig.get_paths()["platlib"])'
/usr/local/lib/python3.10/dist-packages
```

On almalinux:8, python3.6 doesn't have `local`, but python3.8 and python3.11 do have `local` in their sysconfig platlib !

### Information on your system

* **Spack:** 0.22.0.dev0 (90aee11c33a39c6ad62977e45edd1f87528187df)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-skylake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by Jordan474** [comment]
This is a duplicate of  #37632, #32662 and [more](https://github.com/spack/spack/issues/37632#issuecomment-1545911566).
